namesys/publisher.go

```diff
@@ -150,21 +150,36 @@ func PutRecordToRouting(ctx context.Context, k ci.PrivKey, value path.Path, seqn
 		entry.Ttl = proto.Uint64(uint64(ttl.Nanoseconds()))
 	}
 
-	errs := make(chan error, 2)
+	var errorChanSize int
+	var errorChan chan error
 
-	go func() {
-		errs <- PublishEntry(ctx, r, ipnskey, entry)
-	}()
+	// Attempt to extract the public key from the ID
+	var extractedPublicKey = peer.ExtractPublicKey()
+
+	if extractedPublicKey == nil {
+		errorChanSize = 2 // IPNS and public key
+	} else {
+		errorChanSize = 1 // IPNS only
+	}
```

Comment:
This isn't *strictly* necessary as over-allocating isn't really an issue (but it's up to you).